lib/ui/surveys/survey_view.dart

```diff
@@ -0,0 +1,109 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:survey_flutter_ic/gen/assets.gen.dart';
+import 'package:survey_flutter_ic/model/survey_model.dart';
+import 'package:survey_flutter_ic/theme/dimens.dart';
+import 'package:survey_flutter_ic/widget/white_right_arrow_button.dart';
+
+class SurveyView extends ConsumerStatefulWidget {
+  final List<SurveyModel> surveys;
+  final Function(int) onPageChanged;
+  final Function(SurveyModel) onSurveySelected;
+
+  const SurveyView({
+    super.key,
+    required this.surveys,
+    required this.onPageChanged,
+    required this.onSurveySelected,
+  });
+
+  @override
+  ConsumerState<SurveyView> createState() => _SurveyViewState();
+}
+
+class _SurveyViewState extends ConsumerState<SurveyView> {
+  final PageController _pageController = PageController();
+
+  @override
+  Widget build(BuildContext context) {
+    return Stack(
+      children: [
+        PageView.builder(
+          controller: _pageController,
+          onPageChanged: (index) {
+            widget.onPageChanged.call(index);
+          },
+          itemCount: widget.surveys.length,
+          itemBuilder: (BuildContext context, int index) {
```

Comment:
@hoangnguyen92dn We dont need to use the `context` here. It would be better if we create another function name `Widget _pageItemBuilder` and put the implementation inside it 🤔

Reply:
@doannimble So it should be like this? 🤔 

```dart
          itemBuilder: (_, index) {
            return _pageItemBuilder(widget.surveys[index]);
          },
```